api/src/opentrons/protocol_api/validation.py

```diff
@@ -162,6 +172,22 @@ def ensure_lowercase_name(name: str) -> str:
     return name.lower()
 
 
+def ensure_definition_is_adapter(definition: LabwareDefinition) -> None:
+    """Ensure that one of the definition's allowed roles is `adapter`."""
+    if LabwareRole.adapter not in definition.allowedRoles:
+        raise LabwareDefinitionIsNotAdapterError(
+            f"Labware {definition.parameters.loadName} does not defined as an adapter."
+        )
+
+
+def ensure_definition_is_labware(definition: LabwareDefinition) -> None:
+    """Ensure that one of the definition's allowed roles is `labware` or that that field is empty."""
+    if definition.allowedRoles and LabwareRole.labware not in definition.allowedRoles:
+        raise LabwareDefinitionIsNotLabwareError(
+            f"Labware {definition.parameters.loadName} is not defined as a labware."
+        )
```

Comment:
Just an error message suggestion. Otherwise it sounds like this labware definition is missing entirely.

```suggestion
        raise LabwareDefinitionIsNotLabwareError(
            f"Labware {definition.parameters.loadName} is an adapter,"
            f" not a normal labware."
        )
```